fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java

```diff
@@ -296,13 +296,51 @@ protected void lockArchivalGroupResourceFromParent(final Transaction tx,
         if (parentId != null && !parentId.isRepositoryRoot()) {
             final var parentHeaders = pSession.getHeaders(parentId, null);
             if (parentHeaders.isArchivalGroup()) {
-                tx.lockResource(parentId);
+                tx.lockResourceAndGhostNodes(parentId);
             } else if (parentHeaders.getArchivalGroupId() != null) {
-                tx.lockResource(parentHeaders.getArchivalGroupId());
+                tx.lockResourceAndGhostNodes(parentHeaders.getArchivalGroupId());
```

Comment:
I'm not sure if we need to look for ghosts here either. If a parent exists and we lock it, I don't think we care if the parent has any ghosts in its ancestry, do we? Don't we only care about ghosts between the child and its parent?